docs/DesignDocs/edl_security_and_portability.md

```diff
@@ -0,0 +1,118 @@
+EDL security and portability enhancements
+=========================================
+
+Motivation
+----------
+
+This document proposes EDL enhancements that will:
+
+- Improve portability to TrustZone
+- Mitigate security vulnerabilities
+
+This feature is needed by TrustZone to ensure that applications developed with
+EDL will work on TrustZone platforms. Currently the use of host shared memory
+is problematic for TrustZone, which utilizes a different shared memory model.
+SGX *implicitly* shares *all* host memory with the enclave, whereas, TrustZone
+*explicitly* shares *some* host memory with the enclave. Therefore we
+identify three memory models.
+
+- **Implicit-all** -- implicitly shares all host memory (SGX)
+- **Explicit-some** -- explicitly shares some host memory (TrustZone)
+- **None** -- does not share any host memory (future TEEs)
+
+In TrustZone, developers may explicitly share designated memory regions. It
+is not possible to share all host memory (stack, data, heap, etc.) as it is
+in SGX.
```

Comment:
It is technically possible to map the host's memory into a TrustZone enclave. One will run into limitations imposed by OP-TEE's design, but TrustZone does indeed allow it. In other words, one could take the VA range underlying the host, translate that into a list of physical pages, and map them into secure user-mode.